idea/src/org/jetbrains/kotlin/idea/inspections/PropertyVarInspection.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.IntentionWrapper
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import org.jetbrains.kotlin.idea.quickfix.ChangeVariableMutabilityFix
+import org.jetbrains.kotlin.idea.references.mainReference
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.anyDescendantOfType
+
+class PropertyVarInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
+        propertyVisitor(fun(property: KtProperty) {
+            if (property.isLocal || !property.isVar || property.initializer == null || property.setter != null) return
+            val getter = property.getter ?: return
+            if (getter.hasBackingFieldReference()) return
+            holder.registerProblem(
+                property.valOrVarKeyword,
+                "Property setter does not influence its getter result",
+                ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+                IntentionWrapper(
+                    ChangeVariableMutabilityFix(property, makeVar = false, deleteInitializer = true),
+                    property.containingFile
+                )
+            )
+        })
+
+    private fun KtPropertyAccessor.hasBackingFieldReference(): Boolean {
+        val bodyExpression = this.bodyExpression ?: return false
+        val p = this.property
+        return bodyExpression.isBackingField(p) || bodyExpression.anyDescendantOfType<KtNameReferenceExpression> { it.isBackingField(p) }
```

Comment:
There is `BindingContext.BACKING_FIELD_REQUIRED`, please use it and not a raw text check

Reply:
`context[BindingContext.BACKING_FIELD_REQUIRED, propertyDescriptor]` returns `true` in this case:
```kotlin
class Test {
    var foo = 1
        get() = 1
}
```

Reply:
I want to report in following case:
```kotlin
    var foo = 1
        get() = 1
```

I don't want to report in following case:
```kotlin
    var foo: Int = 1
        get() = field
```

`BindingContext.BACKING_FIELD_REQUIRED` returns `true` in both cases, so I do not know how to check property that has backing field reference using `BindingContext.BACKING_FIELD_REQUIRED`.

Reply:
Yes, understood. So you should  both check BACKING_FIELD_REQUIRED and check we do not reference it in getter. Ok.